src/features/exploratory_data_analysis_helpers.py

```diff
@@ -0,0 +1,113 @@
+import re
+from typing import Dict, List, Tuple
+
+import collections
+import pandas as pd
+
+
+def density_of_curse_words_in_sentence(tweet: str) -> Dict[str, float]:
+    """Returns the density of top 20 curse words, taken from Wang, Wenbo,  et  al.
+    Cursing  in english on  twitter."
+    The method needs the punctuation to be removed.
+    Args:
+        tweet (str) : the tweet to be counted.
+    Returns:
+        density (dict) : the curse words and their densities.
+    """
+    curse_words = [
+        "fuck",
+        "shit",
+        "ass",
+        "bitch",
+        "nigga",
+        "hell",
+        "whore",
+        "dick",
+        "piss",
+        "pussy",
+        "slut",
+        "puta",
+        "tit",
+        "damn",
+        "fag",
+        "cunt",
+        "cum",
+        "cock",
+        "blowjob",
+    ]
+
+    # here we are going to use above words as roots in dictionary and then
+    # as dictionary value add them and their plurals in order to make magic happen
+    # I'm just adding plural but you can easily extend it with synonyms and such
+
+    curse_roots = {
+        curse_word: [curse_word, f"{curse_word}s"] for curse_word in curse_words
+    }
+
+    # now we create look_up dictionary which is a reverse of above (all values become
+    # keys, and key become values)
+    lookup = {}
+    for key, values in curse_roots.items():
+        for value in values:
+            lookup[value] = key
+    # here we add counter
+    counts = {curse: 0.0 for curse in curse_words}
+
+    #####
+    tweet_words = tweet.lower().split(" ")
+    # cleaning up white space
+    tweet_words = [tweet_word.strip() for tweet_word in tweet_words]
```

Comment:
We could perhaps use tokenization and lemmatization from one of the nlp packages. It would remove need for storing plurals of curse words and solve problem with words not being detected if they have punctuation signs